### PR TITLE
FIP-46 make walk time configurable for the node

### PR DESCRIPTION
**Pull request by edrotthoff**
FIP-46 make walk time configurable for the node

### CODE REVIEW

File: plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp [hunk 1]
```diff
@@ -48,7 +48,21 @@ namespace eosio {
...
+        /*
+         * These 2 constants are added to provide those who host an API node the ability to modify these values
+         * if there is ever a need to provide API nodes increased time for processing of table information within read only getters.
+         * In FIO we provide 1 second of processing time to an API node for the processing of secondary index information,
+         * this provides API nodes the ability to return reliable results when tables contain up to tens of thousdands of rows per secondary index.
+         * EOSIO historically used a single constant WALKTIME for both primary key and secondary key reads, which were both set to 100 milliseconds, this
+         * limited the number of rows per secondary index to something well under 5k rows per secondary index for reliable results in
+         * a read only getter. SEE EOS ISSUE #3965 https://github.com/EOSIO/eos/issues/3965
+         */
+        const static int SECONDARY_INDEX_MAX_READ_TIME_MICROSECONDS = 1000 * 100; //1 second read time permitted on secondary indices
```
Comment (ericbutz) [inline plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp:62]: The math is 1000 * 100 Microseconds which = 100 milliseconds, but the comment says 1 second which would be 1000 milliseconds?

**Review by ericbutz — CHANGES_REQUESTED** [review]

File: plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp (hunk 1 above)
Comment (edrotthoff) [reply plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp:62]: the comment should be updated I think...it should read .1 second......
theres are two courses of action here, make the timeout to be one second as per the comment, or update the incorrect comment (we had talked about making the timeout to be 10X previous, which is why its set in code at 1000 *100

**Review by adsorptionenthalpy — APPROVED** [review]

File: plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp (hunk 1 above)
Comment (ericbutz) [reply plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp:62]: I am good with updating the comments. Note that the line 63 comment will need to be updated as well.

Comment (edrotthoff) [reply plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp]: comments have been updated, please close the convo if this is all completed to your satisfaction...

**Review by ericbutz — APPROVED** [review]

**Pull request merged**